### DISCUSSION TITLE
Neoteric expressions

### DISCUSSION THREAD

**Discussion by bodemann** [mattwparas/steel#115] [Ideas]
Have you considered allowing Neoteric expressions? 

There is a bit of an discussion regarding syntax at https://news.ycombinator.com/item?id=38508779 
I saw you commenting in the thread but not regarding this topic. I may have overlooked it, though.

Neoteric expression are presented e.g. is this video. https://www.youtube.com/watch?v=MHDmVRU4fqw 
Also relevant: https://readable.sourceforge.io/

I assume this would raise the acceptance level quite a bit. Especially when it comes to integration in a project like helix.

**Comment by mattwparas** [comment]
When you say raise the acceptance level - do you mean make it more acceptable or less acceptable?

Whether I've considered it - I have not really. There is a lot of interesting work with respect to alternative s expression style syntax, in particular I've been following the Rhombus project that the Racket people have been working on.

I personally quite like s expressions, and don't particularly find value in fighting in the syntax wars in the comments, hence my avoidance of that topic in the thread :smile: . As long as the expression can be losslessly compiled to an s expression equivalent so that macros work the same way, I am not opposed to an alternative syntax. I wouldn't prioritize it, but wouldn't immediately say no to it if that makes sense.

What would be nice is if the alternative syntax could be implemented entirely in user space - Racket has a very pleasant functionality with the `#lang` ability to control how the file is parsed entirely via macros. At the moment I have bigger fish to fry, but it would be a nice end goal to have the kind of flexibility in the macro system to have an alternative syntax like the ones you've provided.
